### PR TITLE
Implement GET teams/{team_id}/stats - apiv4

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
- Added ```GET teams/{team_id}/stats``` endpoint

#### Ticket Link
GitHub issue: #5408 

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Added API documentation (required for all new APIs) ([mattermost-api-reference/pull/99](https://github.com/mattermost/mattermost-api-reference/pull/99))
- [x] All new/modified APIs include changes to the drivers

### CODE REVIEW

File: api4/api.go [hunk 1]
```diff
@@ -100,6 +101,7 @@ func InitApi(full bool) {
 	BaseRoutes.TeamByName = BaseRoutes.Teams.PathPrefix("/name/{team_name:[A-Za-z0-9_-]+}").Subrouter()
 	BaseRoutes.TeamMembers = BaseRoutes.Team.PathPrefix("/members").Subrouter()
 	BaseRoutes.TeamMember = BaseRoutes.TeamMembers.PathPrefix("/{user_id:[A-Za-z0-9]+}").Subrouter()
+	BaseRoutes.TeamStats = BaseRoutes.Team.PathPrefix("/stats").Subrouter()
```
Comment (jwilander) [inline api4/api.go]: For this route can we actually not define it here, but in the `InitTeam()` function in `api4/team.go`?

File: api4/team.go [hunk 2]
```diff
@@ -21,7 +21,7 @@ func InitTeam() {
 	BaseRoutes.Team.Handle("", ApiSessionRequired(getTeam)).Methods("GET")
 
 	BaseRoutes.TeamMember.Handle("", ApiSessionRequired(getTeamMember)).Methods("GET")
-
+	BaseRoutes.TeamStats.Handle("", ApiSessionRequired(getTeamStats)).Methods("GET")
```
Comment (jwilander) [inline api4/team.go]: Per my above comment, this would become `BaseRoutes.Team.Handle("/stats", ApiSessionRequired(getTeamStats)).Methods("GET")`

**Comment by saturninoabril** [comment]
@jwilander noted. please see changes made. thanks!

**Comment by jwilander** [comment]
Thanks @saturninoabril, looks good!

**Pull request merged**